mllib/src/main/scala/org/apache/spark/ml/regression/LinearRegression.scala

```diff
@@ -645,3 +666,179 @@ private class LeastSquaresCostFun(
     (leastSquaresAggregator.loss + regVal, new BDV(totalGradientArray))
   }
 }
+
+/**
+ * HuberCostFun implements Breeze's DiffFunction[T] for Huber cost as used in Robust regression.
```

Comment:
nit: surround references to code in `[[DiffFunction[T]]]` to have docs generate links